### ISSUE TITLE
Function runtime crash on ~2 with Node.js

### ISSUE THREAD

**Issue by khaosdoctor** [Azure/azure-functions-host#4873]
#### Investigative information

Please provide the following:

- Timestamp: 2019-09-03T01:13:46.880Z
- Function App version (1.0 or 2.0): 2.0
- Function App name: Dignify
- Function name(s) (as appropriate): FnGetPaymentHistory, FnPostLogin
- Invocation ID: Never invoked
- Region: Brazil South

#### Repro steps

Provide the steps required to reproduce the problem:

1. Create project on VSCode using TypeScript
2. Create function which connects to external databases (using shared code to connect to, for instance, a Mongo Database)
3. `host.json` file:

```json
{
  "version": "2.0",
  "watchDirectories": ["shared"]
}
```

4. Publish function
5. Check configuration `FUNCTIONS_EXTENSION_VERSION`, the value should be `~2`
6. Check configuration `FUNCTIONS_WORKER_RUNTIME`, should be `node`
7. `WEBSITE_NODE_DEFAULT_VERSION` should be `10.6.0`
8. Open Function app in azure portal
9. Get error:

![image](https://user-images.githubusercontent.com/3200560/64184012-2aeecd00-ce41-11e9-8555-2450e69af5ac.png)

#### Expected behavior

After publishing the function, the runtime should start

#### Actual behavior

The runtime fails to start, therefore the function is never executed

#### Known workarounds

If I change `FUNCTIONS_EXTENSION_VERSION` to `~1` and restart the app, things start working again. But then I get compatibility issues when deploying from VSCode

#### Related information 

Provide any related information 

* Related issues: #3355, #3578, #3947
* Programming language used: NodeJS V10
* Links to source: None
* Bindings used: Below there's the `function.json`

```json
{
  "bindings": [{
      "authLevel": "function",
      "type": "httpTrigger",
      "direction": "in",
      "name": "req",
      "methods": [
        "get"
      ],
      "route": "payments/history"
    },
    {
      "type": "http",
      "direction": "out",
      "name": "$return"
    }
  ],
  "scriptFile": "../dist/FnGetPaymentHistory/index.js"
}
```

<details>
<summary>Source For GetPaymentHistory</summary>

```ts
import boom from 'boom'
import { config } from '../shared/app-config'
import { Response } from '../shared/utils/Response'
import { Context, HttpRequest } from "@azure/functions"
import { createConnection } from '../shared/data/connection/mongoConnection'
import { PaymentRepository } from '../shared/data/repositories/PaymentRepository'

export default async function getPaymentHistory (context: Context, req: HttpRequest): Promise<any> {
  const res = new Response()
  try {
    const { page, size } = req.query
    if (page && size && (isNaN(parseInt(page)) || isNaN(parseInt(size)))) return res.status(422).json(boom.badData('Query values are not numbers').output.payload)

    const connection = await createConnection(config.database.mongodb)
    const paymentRepository = new PaymentRepository(connection.db)
    const result = await paymentRepository.getHistory(parseInt(page) || 1, parseInt(size) || 50)
    return res.status(200)
      .headers({
        'x-range-from': result.range.from,
        'x-range-to': result.range.to,
        'x-range-total': result.total,
        'x-range-size': result.count
      })
      .json(result.documents)
  } catch (error) {
    context.log.error({ params: req.params, query: req.query }, error)
    return res.status(500).json(boom.internal().output.payload)
  }
}
```
</details>

<details>
<summary>Source For Response Utility</summary>

```ts
export interface ResponseState {
  body?: string
  status?: number
  headers: {
    [key: string]: string | boolean | number
  }
}

export class Response {
  private state: ResponseState = {
    headers: {}
  }
  json (payload: any) {
    this.state.body = JSON.stringify(payload)
    this.state.headers['Content-Type'] = 'application/json'
    return this.state
  }

  status (number: number) {
    this.state.status = number
    return this
  }

  headers (headers: ResponseState['headers']) {
    this.state.headers = headers
    return this
  }

  send (payload: any) {
    this.state.body = payload
    return this
  }

  end () {
    const state = Object.assign({}, this.state)
    this.state = {
      headers: {}
    }
    return state
  }
}
```
</details>

**Comment by glaucia86** [comment]
cc: @gzuber one more issue... and this one seems to be an inconsistency problem

**Comment by khaosdoctor** [comment]
So... I guess it's working now. However, I don't know exactly what I did to solve the problem.

I cloned an empty repository and started a new project from scratch, then I compared `package.json`, `tsconfig.json` and other key files to my own files. My `package.json` had, besides the dependencies, an extra key for `main` which was set to `dist/index.js` (a leftover from the previous version of the app, so the file did not exist). I removed this key and removed the `excludes`, `incremental` keys from `tsconfig` and added `rootDir` of `.`.

After that I completely removed the resource group assigned to this function app and hit deploy on VSCode selecting the "advanced" option. I selected a Linux distribution and a ConsumptionPlan instead of an AppServicePlan. Then I rolled out the deploy and it started working, somehow.

The code remained the same, but it started working now with the `~2` flag set.
